forge/routes/api/device.js

```diff
@@ -334,36 +335,52 @@ module.exports = async function (app) {
     }, async (request, reply) => {
         let sendDeviceUpdate = false
         const device = request.device
-        if (request.body.instance !== undefined) {
+        if (request.body.instance !== undefined || request.body.application !== undefined) {
             // ### Add/Remove device to/from project ###
 
-            if (request.body.instance === null) {
-                // ### Remove device from project ###
+            const assignTo = request.body.instance ? 'instance' : (request.body.application ? 'application' : null)
 
-                if (device.Project !== null) {
-                    const oldProject = device.Project
-                    // unassign from project
-                    await device.setProject(null)
+            if (!assignTo) {
+                // ### Remove device from application/project ###
+                const commonUpdates = async () => {
                     // Clear its target snapshot, so the next time it calls home
                     // it will stop the current snapshot
                     await device.setTargetSnapshot(null)
                     sendDeviceUpdate = true
+                    // disable developer mode
+                    device.mode = 'autonomous'
+                    await device.save()
+                }
+
+                if (device.Application !== null) {
+                    // const oldApplication = device.Application
+                    // unassign from application
+                    await device.setApplication(null)
+                    await commonUpdates()
+                    // TODO
```

Comment:
Are these genuine todos or left overs?

Reply:
Future TODOs - in line with other TODOs already in surrounding tests.

TBH, they can be removed as, strictly speaking,  they are really separate unit test related to audit logging.